qualtran/_infra/controlled_test.py

```diff
@@ -374,7 +377,7 @@ def build_composite_bloq(self, bb: 'BloqBuilder') -> Dict[str, 'SoquetT']:
         and_ctrl = [bb.add(one_or_zero[cv]) for cv in self.and_ctrl]
 
         ctrl_soqs = bb.add_d(cbloq, **ctrl_soqs, ctrl=and_ctrl)
-        out_soqs = [*ctrl_soqs.pop('ctrl'), ctrl_soqs.pop('target')]
+        out_soqs = np.asarray([*ctrl_soqs.pop('ctrl'), ctrl_soqs.pop('target')])
```

Comment:
here's another philosophical question. Really, the return value is supposed to be a numpy array. But we've coded the library to still work if it's a list (by sprinkling defensive `np.asarray` calls throughout). Is it worth keeping some of these badly-typed examples in the tests to keep those code paths exercised?

Reply:
I think that if you want to officially support Lists, then the function declaration should state it.